src/daemon/ca/certauth.rs

```diff
@@ -1841,21 +1845,40 @@ impl CertAuth {
         Ok(update_events)
     }
 
-    /// Verifies whether the update needs to be applied
+    /// Verifies whether the update is allowed and needs to be applied.
     ///
     /// The update does not need to be applied if there would be no change in
     /// the configured AspaDefinition. I.e. this gives us idempotence and e.g. allows
     /// an operator just issue a command to add a provider for a customer ASN, and
     /// if it was already authorised then no work is needed.
-    fn updated_needed(&self, customer: AspaCustomer, update: &AspaProvidersUpdate) -> KrillResult<bool> {
-        if update.is_empty() {
+    fn updated_allowed_and_needed(&self, customer: AspaCustomer, update: &AspaProvidersUpdate) -> KrillResult<bool> {
```

Comment:
s/updated/update/ ?

Reply:
The function argument is the 'update', but the result is (potentially) something 'updated'. So, leaving this as-is.